packages/medical-domain/src/domains/repositories/statistics/glycemia-statistics.service.ts

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { MS_IN_MIN, MS_IN_DAY, diffDays } from '../time/time.service'
+import type Cbg from '../../models/medical/datum/cbg.model'
+import type Smbg from '../../models/medical/datum/smbg.model'
+import type { BgBounds, AverageGlucoseStatistics, CbgRangeStatistics, CoefficientOfVariationStatistics, GlucoseManagementIndicatoStatistics, SensorUsageStatistics, StandardDevStatistics } from '../../models/statistics/glycemia-statistics.model'
+import { ClassificationType } from '../../models/statistics/enum/bg-classification.enum'
+import CbgService, { convertBG } from '../medical/datum/cbg.service'
+import SmbgService from '../medical/datum/smbg.service'
+import type DateFilter from '../../models/time/date-filter.model'
+import { type WeekDaysFilter, defaultWeekDaysFilter } from '../../models/time/date-filter.model'
+import type Bg from '../../models/medical/datum/bg.model'
+import { type BgUnit, MGDL_UNITS, MMOLL_UNITS } from '../../models/medical/datum/bg.model'
+
+export function classifyBgValue(bgBounds: BgBounds, bgValue: number, classificationType: ClassificationType): keyof CbgRangeStatistics {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Shouldn't this method replace the one in `packages/viz/src/utils/bloodglucose.js`?

Reply:
Yes... I didn't check the usage after migrating it

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏I feel like now you can remove the method from `packages/viz/src/utils/bloodglucose.js`?

Reply:
Forgot one file in my last commit... Now done.